src/context/toast-context.js

```diff
@@ -0,0 +1,21 @@
+import { useState, createContext, useContext, useEffect } from "react";
+
+const ToastContext = createContext();
+
+const ToastProvider = ({ children }) => {
+  const [toastData, setToastData] = useState({
+    showToast: false,
+    toastType: "",
+    toastMesaage: "",
```

Comment:
suggestion: on naming, ` toast ` gets repeated on object and property as well eg. ` toastData.toastType ` can be simplified to ` toast.type `